lib/yunohost/tools.py

```diff
@@ -410,11 +413,25 @@ def tools_upgrade(auth, ignore_apps=False, ignore_packages=False):
     """
     from yunohost.app import app_upgrade
 
-    failure = False
-
     # Retrieve interface
     is_api = True if msettings.get('interface') == 'api' else False
 
+    # Special case for v2.2 to v2.4 upgrade
+    if not v24 and not is_api:
+        try:
+            i = msignals.prompt(m18n.n('upgrade_24_confirm',
+                                       answers='y/N'))
+        except NotImplemented:
+            pass
+        else:
+            if i == 'y' or i == 'Y':
```

Comment:
You can do a `i in ('y', 'Y')' or even a `i.lower() == 'y'` (if you are sure that `i` will have a `.lower` method).